### PR TITLE
Fix pixelation to only happen when treating Markers.

### PR DESCRIPTION
**Pull request by miscco**
Grml,

another small thing i noticed, with my own plots. When we have a mixed linestyle aka '-o' then pixelation will create new lines as it doesnt check for the line segments.

I have to add Test for that

### CODE REVIEW

**Comment by miscco** [comment]
So i am currently not 100% sure, how my data triggered this. 

My guess is, that if you have partially overlapping plots with mixed linestyle, aka '-o', than if the second plot is a partial! subset of the first one, it can happen, that we deleted everyting in between two points generating a new line.

**Comment by miscco** [comment]
Should i put the above code into a separate test or i there a test one could modify?

**Comment by okomarov** [comment]
It is really hard to follow what you mean without any visual example.

Your current edit will introduce another problem. 

The following:

    plot1:10,1:10,'-o')

will be simplified to 2 points...which is not what I want since I expect 10 markers.

**Comment by miscco** [comment]
Have you pulled the latest commit 80823a0? 

Because it doesnt remove the markers anymore. As a graphical example consider a plot that returns to a coordinate:
```matlab
xdata = [0,1,2,2,1,0];
ydata1 = [1, 0, 1, -1, 0 ,-1];
plot(xdata, ydata1, '-o');
```
example_develop

This will be cleaned by pixelation to the following

example_develop_cleaned

The reason is the reoccuring marker and the fact that the previous code doesnt check for markers in the lineSimplify branch.

Previously this problem wouldnt occur, as pixelate only removed neigbourghs

**Comment by okomarov** [comment]
No, I was just looking at the code. You raise quite a valid point.

File: src/cleanfigure.m [hunk 1]
```diff
@@ -410,11 +410,13 @@ function simplifyLine(meta, handle, targetResolution)
 
     % If the path has markers, perform pixelation instead of simplification
     hasMarkers = ~strcmpi(get(handle,'Marker'),'none');
-    if hasMarkers
+    hasLines   = ~strcmpi(get(handle,'LineStyle'),'none');
+    if hasMarkers && ~hasLines
         % Pixelate data at the zoom multiplier
         mask      = pixelate(xData, yData, xToPix, yToPix);
         id_remove = find(mask==0);
-    else
```
Comment (miscco) [inline src/cleanfigure.m]: I dont know what is prefered here 
```
elseif hasLines && ~hasMarkers

or

end
if hasLines && ~hasMarkers
```

Comment (egeerardyn) [inline src/cleanfigure.m]: I would go for 
```matlab
if hasMarkers && ~hasLines
    ...
elseif ~hasMarkers && hasLines
    ...
end
```

that makes it clearer that both are mutually exclusive (while independent `if` statements need not be).

**Comment by miscco** [comment]
@matlab2tikzBot: Please test this.

**Comment by miscco** [comment]
@matlab2tikzBot: Please test this.

**Comment by miscco** [comment]
It seems like travis is down again :disappointed:

**Comment by miscco** [comment]
@matlab2tikzBot: Please test this.

**Comment by egeerardyn** [comment]
@miscco : to retest on Travis, you have to click on "Details" next to the travis results (this brings you to the relevant Travis build page). If you are logged in, normally you should see a "redo" arrow on the right side of the page. If you click on this, that will redo the build. The same also works on Jenkins by the way: there is also a button to redo the last build without those magic phrases.

In any case, I reran this on Travis, but it seems you will need to pull in develop into your branch to fix this.

**Comment by miscco** [comment]
Yeah, however, my working directory is rather messy right now from the grid rework, so it will take some time till i can do a checkout

**Comment by egeerardyn** [comment]
You might want to have a look at [`git stash`](https://git-scm.com/book/en/v1/Git-Tools-Stashing) :smile_cat:

**Comment by miscco** [comment]
How is the praxis here about accepting a PR? Normally i am very reluctant to accept my own stuff.

**Comment by egeerardyn** [comment]
@miscco: [Typical approach](https://github.com/matlab2tikz/matlab2tikz/blob/develop/CONTRIBUTING.md) is that every PR should get a :+1: by at least another developer and that all comments have been addressed before merging. Typically, the one who writes ":+1:" will merge it then. Otherwise, we often write ":+1:, feel free to merge when $condition".

For secondary stuff (documentation, typos, fixing bugs in Travis or our testing framework, ...) where the fix is just obvious, also open a PR, wait a reasonable while (about an hour or more) but feel free to merge it when it has some urgency.
E.g. yesterday when fixing the Travis problems, I wasn't waiting for a ":+1:" for this reason: it blocked all other PRs and the fix was obvious and tested.

I will review this PR tonight, since this touches main functionality.

**Comment by egeerardyn** [comment]
I have checked this on Octave 3.8.2, R2014A and R2014B: with this PR applied and the latest develop (3323de6138b23790d1d7172f60bcd196b2244359).

Only `ACID(8)` gives different results on R2014A (HG1):
**before**
BEFORE R2014A
**after**
AFTER R2014A

The position of the labels, I suspect, is a MATLAB thingy that we have little control over. Anyhow, since the test is unreliable, I don't mind this too much (although it is something we should inspect).
In the current develop, `cleanfigure` also seems to have a noticeable (=bad) effect on this plot in R2014a.

In Octave 3.8, there are more differences `[19,20,28,32,33,36,61,68,77,79,92,94]` in the output TikZ code:
**before**
test19-converted
test20-converted
test28-converted
test32-converted
test33-converted
test36-converted
test61-converted
test68-converted
test77-converted
test79-converted
test92-converted
test94-converted

**after**
test19-converted
test20-converted
test28-converted
test32-converted
test33-converted
test36-converted
test61-converted
test68-converted
test77-converted
test79-converted
test92-converted
test94-converted

At first glance, I don't see any differences, but `compare` thinks otherwise...

**Comment by okomarov** [comment]
I can see that contours are flipped. Look at the inner small ones.

**Comment by miscco** [comment]
@egeerardyn Have you checked the diff. Usually compare takes 3 arguments:
```
compare before after diff
```

That should help finding the differences

**Comment by egeerardyn** [comment]
I threw away the diffs yesterday, because I was a bit tired. I re-ran the `compares` and this is the result:

test19-converted
test20-converted
test28-converted
test32-converted
test33-converted
test36-converted
test61-converted
test68-converted
test77-converted
test79-converted
test94-converted

**Comment by miscco** [comment]
That actually seems like there is no difference at all. 

What kind of metric did you use. Here maybe AE might be most suited

**Comment by egeerardyn** [comment]
@miscco both PSNR and AE perform poorly here. For most figures, it seems to use a different color scheme, some have nudged the figure around a bit, only in the bar+line there is a difference (nothing related to the simplification, I guess) and one figure had different size.
This is the output I got:
```
test19-converted.png PNG 1261x795 1261x795+0+0 8-bit sRGB 49.8KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test19-converted.png PNG 1261x795 1261x795+0+0 8-bit sRGB 48.7KB 0.000u 0:00.000
Image: test19-converted.png
  Channel distortion: AE
    red: 5220
    green: 5220
    blue: 5220
    alpha: 5220
    all: 5220
test19-converted.png=>/Users/egon/Desktop/diffs/test19-converted.png PNG 1261x795 1261x795+0+0 8-bit sRGB 0.000u 0:00.000
test20-converted.png PNG 1351x791 1351x791+0+0 8-bit sRGB 19.7KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test20-converted.png PNG 1351x791 1351x791+0+0 8-bit sRGB 22KB 0.000u 0:00.000
Image: test20-converted.png
  Channel distortion: AE
    red: 16026
    green: 16026
    blue: 16026
    alpha: 16026
    all: 16026
test20-converted.png=>/Users/egon/Desktop/diffs/test20-converted.png PNG 1351x791 1351x791+0+0 8-bit sRGB 134c 0.000u 0:00.000
test28-converted.png PNG 1272x791 1272x791+0+0 8-bit sRGB 13.7KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test28-converted.png PNG 1272x791 1272x791+0+0 8-bit sRGB 13.7KB 0.000u 0:00.000
Image: test28-converted.png
  Channel distortion: AE
    red: 4613
    green: 4613
    blue: 4613
    alpha: 4613
    all: 4613
test28-converted.png=>/Users/egon/Desktop/diffs/test28-converted.png PNG 1272x791 1272x791+0+0 8-bit sRGB 98c 0.000u 0:00.000
test32-converted.png PNG 1239x791 1239x791+0+0 8-bit sRGB 19.3KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test32-converted.png PNG 1239x791 1239x791+0+0 8-bit sRGB 19.2KB 0.000u 0:00.000
Image: test32-converted.png
  Channel distortion: AE
    red: 21034
    green: 21034
    blue: 21034
    alpha: 21034
    all: 21034
test32-converted.png=>/Users/egon/Desktop/diffs/test32-converted.png PNG 1239x791 1239x791+0+0 8-bit sRGB 0.000u 0:00.000
test33-converted.png PNG 1239x791 1239x791+0+0 8-bit sRGB 39.6KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test33-converted.png PNG 1239x791 1239x791+0+0 8-bit sRGB 39.8KB 0.000u 0:00.000
Image: test33-converted.png
  Channel distortion: AE
    red: 23996
    green: 23996
    blue: 23996
    alpha: 23996
    all: 23996
test33-converted.png=>/Users/egon/Desktop/diffs/test33-converted.png PNG 1239x791 1239x791+0+0 8-bit sRGB 0.000u 0:00.000
test36-converted.png PNG 1228x791 1228x791+0+0 8-bit sRGB 13.4KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test36-converted.png PNG 1228x791 1228x791+0+0 8-bit sRGB 13.4KB 0.000u 0:00.000
Image: test36-converted.png
  Channel distortion: AE
    red: 15444
    green: 15444
    blue: 15444
    alpha: 15444
    all: 15444
test36-converted.png=>/Users/egon/Desktop/diffs/test36-converted.png PNG 1228x791 1228x791+0+0 8-bit sRGB 145c 0.000u 0:00.000
test61-converted.png PNG 1459x847 1459x847+0+0 8-bit sRGB 21.1KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test61-converted.png PNG 1459x847 1459x847+0+0 8-bit sRGB 23.7KB 0.000u 0:00.000
Image: test61-converted.png
  Channel distortion: AE
    red: 15461
    green: 15461
    blue: 15461
    alpha: 15461
    all: 15461
test61-converted.png=>/Users/egon/Desktop/diffs/test61-converted.png PNG 1459x847 1459x847+0+0 8-bit sRGB 94c 0.000u 0:00.000
test68-converted.png PNG 1217x870 1217x870+0+0 8-bit sRGB 21.1KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test68-converted.png PNG 1217x870 1217x870+0+0 8-bit sRGB 21.1KB 0.000u 0:00.000
Image: test68-converted.png
  Channel distortion: AE
    red: 7836
    green: 7836
    blue: 7836
    alpha: 7836
    all: 7836
test68-converted.png=>/Users/egon/Desktop/diffs/test68-converted.png PNG 1217x870 1217x870+0+0 8-bit sRGB 207c 0.000u 0:00.000
test77-converted.png PNG 1217x791 1217x791+0+0 8-bit sRGB 13.7KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test77-converted.png PNG 1217x791 1217x791+0+0 8-bit sRGB 13.6KB 0.000u 0:00.000
Image: test77-converted.png
  Channel distortion: AE
    red: 403062
    green: 403062
    blue: 403062
    alpha: 403062
    all: 403062
test77-converted.png=>/Users/egon/Desktop/diffs/test77-converted.png PNG 1217x791 1217x791+0+0 8-bit sRGB 106c 0.000u 0:00.000
test79-converted.png PNG 1229x791 1229x791+0+0 8-bit sRGB 15.3KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test79-converted.png PNG 1229x791 1229x791+0+0 8-bit sRGB 15.3KB 0.000u 0:00.000
Image: test79-converted.png
  Channel distortion: AE
    red: 3917
    green: 3917
    blue: 3917
    alpha: 3917
    all: 3917
test79-converted.png=>/Users/egon/Desktop/diffs/test79-converted.png PNG 1229x791 1229x791+0+0 8-bit sRGB 110c 0.000u 0:00.000
test92-converted.png PNG 1230x726 1230x726+0+0 8-bit sRGB 18.9KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test92-converted.png PNG 1230x744 1230x744+0+0 8-bit sRGB 27.9KB 0.000u 0:00.000
compare: image widths or heights differ `test92-converted.png' @ error/compare.c/CompareImageCommand/990.
test94-converted.png PNG 1228x791 1228x791+0+0 8-bit sRGB 39.9KB 0.000u 0:00.000
../../../Octave38-after/data/converted/test94-converted.png PNG 1228x791 1228x791+0+0 8-bit sRGB 38.5KB 0.000u 0:00.000
Image: test94-converted.png
  Channel distortion: AE
    red: 18308
    green: 18308
    blue: 18308
    alpha: 18308
    all: 18308
test94-converted.png=>/Users/egon/Desktop/diffs/test94-converted.png PNG 1228x791 1228x791+0+0 8-bit sRGB 0.000u 0:00.000
```

If you want to play around with the files, I can always package them up in a ZIP file or something.

**Comment by miscco** [comment]
Nah, i dont think this is worth it. What is really confusing me is,that some of the plots dont even have markers, so there should be no difference there

**Comment by miscco** [comment]
Just for my consciousness can you check the tex files?

what i did is:
1. Run testGraphical('output', 'output/PixelateMarkers')
2. Switch branch and run testGraphical('output', 'output/develop')
3. Run the following command
```
for i in data/converted/*.tex; do diff "$i" "../Pixelate_octave/$i" > diff.text; done
```

**Comment by miscco** [comment]
Ping

**Comment by egeerardyn** [comment]
@miscco Have you tried this out on my test server (I sent you SSH login information a while back) to run that script on those environments yourself?

**Comment by miscco** [comment]
@egeerardyn not yet actually, i dont know if i find the time this week

**Comment by egeerardyn** [comment]
@miscco No worries :-) It's just that it's really unlikely that I will find some time the coming weeks (writing a paper and a thesis).

**Comment by miscco** [comment]
Ok i found my error, i used > in the script isntead of >> :cry:

**Comment by miscco** [comment]
Ah there was as always a mistake, it seems ACID(8) does some things too in Matlab2014a too, but they are similar to those in the other PR.
